diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75bc72a..a109fb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,14 +50,46 @@ if(ASIOSDK_DIR)
     target_compile_definitions(juce_plugin_modules PUBLIC JUCE_ASIO=1)
 endif()
 
+# begin JUCEAAP specific -->
+# It is not beautiful, but there does not seem to be any way to add custom include directory for modules.
+include_directories(
+        "${AAP_DIR}/native/androidaudioplugin/core/include"
+        "${AAP_DIR}/native/plugin-api/include"
+        )
+
+juce_add_module(${AAP_JUCE_DIR}/modules/juceaap_audio_plugin_client)
+
+if (ANDROID)
+
+# dependencies
+find_library(log "log")
+find_library(android "android")
+find_library(glesv2 "GLESv2")
+find_library(egl "EGL")
+set(cpufeatures_lib "cpufeatures")
+set(oboe_lib "oboe")
+
+endif (ANDROID)
+# <-- end JUCEAAP specific
+
 target_link_libraries(ChowPhaserMono PRIVATE
     BinaryData
     juce_plugin_modules
+        juceaap_audio_plugin_client
+        ${log}
+        ${android}
+        ${glesv2}
+        ${egl}
 )
 
 target_link_libraries(ChowPhaserStereo PRIVATE
     BinaryData
     juce_plugin_modules
+        juceaap_audio_plugin_client
+        ${log}
+        ${android}
+        ${glesv2}
+        ${egl}
 )
 
 # we need these flags for notarization on MacOS
diff --git a/modules/JUCE b/modules/JUCE
index f00a420..e6ec181 160000
--- a/modules/JUCE
+++ b/modules/JUCE
@@ -1 +1 @@
-Subproject commit f00a420bc348d79c4688b33b2b905b8ca0f25a3a
+Subproject commit e6ec1819ec0a59a7cfe82d8ce72367f64e0a4bb0-dirty
