diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75bc72a..f050942 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,14 +50,47 @@ if(ASIOSDK_DIR)
     target_compile_definitions(juce_plugin_modules PUBLIC JUCE_ASIO=1)
 endif()
 
+# begin JUCEAAP specific -->
+# It is not beautiful, but there does not seem to be any way to add custom include directory for modules.
+include_directories(
+        "${AAP_DIR}/include"
+        )
+
+juce_add_module(${AAP_JUCE_DIR}/aap-modules/aap_audio_processors)
+
+add_compile_definitions([[JUCE_DONT_AUTO_OPEN_MIDI_DEVICES_ON_MOBILE=1]])
+
+if (ANDROID)
+
+# dependencies
+find_library(log "log")
+find_library(android "android")
+find_library(glesv2 "GLESv2")
+find_library(egl "EGL")
+set(cpufeatures_lib "cpufeatures")
+set(oboe_lib "oboe")
+
+endif (ANDROID)
+# <-- end JUCEAAP specific
+
 target_link_libraries(ChowPhaserMono PRIVATE
     BinaryData
     juce_plugin_modules
+        aap_audio_processors
+        ${log}
+        ${android}
+        ${glesv2}
+        ${egl}
 )
 
 target_link_libraries(ChowPhaserStereo PRIVATE
     BinaryData
     juce_plugin_modules
+        aap_audio_processors
+        ${log}
+        ${android}
+        ${glesv2}
+        ${egl}
 )
 
 # we need these flags for notarization on MacOS
diff --git a/modules/JUCE b/modules/JUCE
--- a/modules/JUCE
+++ b/modules/JUCE
@@ -1 +1 @@
-Subproject commit f00a420bc348d79c4688b33b2b905b8ca0f25a3a
+Subproject commit f00a420bc348d79c4688b33b2b905b8ca0f25a3a-dirty
